src/main/java/Containers/Portfolio.java

```diff
@@ -23,67 +23,116 @@ public static Portfolio getInstance() {
         return Portfolio.commonPortfolio;
     }
 
-    // Storage: The UUID of asset and the reference of asset
-    private final HashMap<String, Asset> storage;
+
+
+    // assetList: a list of asset in the portfolio
+    private List<Asset> assetList;
+    // transactionList: a list of transaction in the portfolio
+    private List<Transaction> transactionList;
+    // votingHistory: a history of votes in the portfolio
+    private List<Vote> votingHistory;
+    private double profitability;
 
     private Portfolio() {
-        this.storage = new LinkedHashMap<String, Asset>();
+        this.assetList = new ArrayList<>();
+        this.transactionList = new ArrayList<>();
+        this.votingHistory = new ArrayList<>();
+        this.profitability = 0.0;
+    }
+    // Getter and Setter for profitability
+    public double getProfitability() {
+        return profitability;
+    }
+
+    public void setProfitability(double profitability) {
+        this.profitability = profitability;
+    }
+
+    // Getter and Setter for assets, transactions, and votes
+
+    public List<Asset> getAssetList() {
+        return assetList;
+    }
+
+    public List<Transaction> getTransactionList() {
+        return transactionList;
+    }
+
+    public List<Vote> getVotingHistory() {
+        return votingHistory;
     }
 
     // Add an asset to the system.
     // This method only takes a snapshot of its parameter, the parameter object can be safely modified afterwards.
     public void add(Asset asset0) {
```

Comment:
What happens if asset0 is null? Does this method require that asset0 must not be null? Also, I think you should write a javadoc for this method, since it is a core part of this class.